core/meta/src/TClass.cxx

```diff
@@ -5464,29 +5464,24 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
-   R__LOCKGUARD(gInterpreterMutex);
+   // Load class info if not already loaded and auto-parsing is not suspended
+   if (!fClassInfo && !gInterpreter->IsAutoParsingSuspended()) {
```

Comment:
> BTW, if fClassInfo is being set during the if statement, then the lock is acquired. By the time we can continue, fClassInfo should be either not set or fully set, so the check for early return will take care that we don't redo the work or return an incomplete result.

Unfortunately not accurate.  In the (rare, very rare but real and seen by CMSSW) cases where fClassInfo is partially filled when the if statement happens, then the function returns immediately and then there is (still) a chance that the reading thread will then proceed to try to use this partially filled value of fClassInfo.